front/lib/api/assistant/actions/tables_query.ts

```diff
@@ -69,19 +87,91 @@ export class TablesQueryAction extends BaseAction {
   }
 
   renderForMultiActionsModel(): FunctionMessageTypeModel {
+    const partialOutput: Omit<FunctionMessageTypeModel, "content"> = {
+      role: "function" as const,
+      name: this.functionCallName ?? DEFAULT_TABLES_QUERY_ACTION_NAME,
+      function_call_id: this.functionCallId ?? `call_${this.id.toString()}`,
+    };
+
+    // Unexpected error case -- we don't have any action output.
+    if (!this.output) {
+      return {
+        ...partialOutput,
+        content: "(query failed)",
+      };
+    }
+
     let content = "";
-    content += `OUTPUT:\n`;
 
-    if (this.output === null) {
-      content += "(query failed)\n";
-    } else {
-      content += `${JSON.stringify(this.output, null, 2)}\n`;
+    // Add reasoning if it exists (should always exist).
+    if (typeof this.output.thinking === "string") {
+      content += `Reasoning:\n${this.output.thinking}\n\n`;
+    }
+
+    const query =
+      typeof this.output.query === "string" ? this.output.query : null;
+
+    if (!query) {
+      // Model didn't generate a query, so we don't have any results.
+      content += "No query was executed.\n";
+      return {
+        ...partialOutput,
+        content,
+      };
+    }
+
+    content += `Query:\n${this.output.query}\n\n`;
+
+    const error =
+      typeof this.output.error === "string" ? this.output.error : null;
+
+    if (error) {
+      // Generated query failed to execute.
+      content += `Error:\n${this.output.error}\n\n`;
+      return {
+        ...partialOutput,
+        content,
+      };
+    }
+
+    const hasResultsFile = this.resultsFileId && this.resultsFileSnippet;
+
+    if (!hasResultsFile && !this.output.results) {
+      // We don't have any results -- this is unexpected, we should always
+      // have either an eror or some results (either a file or a `results` prop).
+      content += "No results were returned.\n";
+      return {
+        ...partialOutput,
+        content,
+      };
+    }
+
+    if (hasResultsFile) {
+      const attachment = getTablesQueryResultsFileAttachment({
+        resultsFileId: this.resultsFileId,
+        resultsFileSnippet: this.resultsFileSnippet,
+        output: this.output,
+        includeSnippet: true,
+      });
+      if (!attachment) {
+        throw new Error(
+          "Unexpected: No file attachment for tables query with results file."
+        );
+      }
```

Comment:
Let's use `assert(attachment, "Unexpected: No file attachment for tables query with results file.");`

Reply:
Why ?

Reply:
You save two lines 😅.

Topic for next week Eng meeting but: https://github.com/dust-tt/dust/pull/7922#discussion_r1790461818.

Reply:
Well I have to import assert so I don't really save a line + I find it less clear. I don't need to import a lib to throw an error.

Agreed with @Fraggle  on this one, I don't know about this specific assert but to me assert carries a meaning that it is not meant for production runtime checks (in many languages asserts get stripped out from prod builds)